### ISSUE TITLE
`id-match` should use boolean schema type

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#4155]
The schema of `id-match` is currently:

```js
    {
        "type": "string"
    },
    {
        "type": "object",
        "properties": {
            "properties": {
                "enum": [true, false]
            }
        }
    }
```

I think it should be:

```js
    {
        "type": "string"
    },
    {
        "type": "object",
        "properties": {
            "properties": {
                "type": "boolean"
            }
        }
    }
```

The difference being use of the boolean type rather than an enum.  I don't think it causes any problems the way it is, but it should probably be corrected just for the sake of correctness and consistency.

**Comment by nzakas** [comment]
Working on this.
